### PR TITLE
More compliance checks cleanups

### PR DESCRIPTION
**Pull request by carlescufi**
See commit messages.

CC @cfriedt

### CODE REVIEW

**Review by cfriedt — DISMISSED** [review]

File: scripts/ci/check_compliance.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ class DevicetreeBindingsCheck(ComplianceTest):
     """
     Checks if we are introducing any unwanted properties in Devicetree Bindings.
     """
-    name = "Devicetree Bindings"
+    name = "DevicetreeBindings"
```
Comment (gmarull) [inline scripts/ci/check_compliance.py:225]: we should be able to pass `-m "Devicetree Bindings"`, but I agree it's better to not use spaces.

File: scripts/ci/check_compliance.py [hunk 2]
```diff
@@ -30,8 +30,6 @@
     # Propagate this decision to child processes.
     os.environ['ZEPHYR_BASE'] = ZEPHYR_BASE
 
-    print(f'ZEPHYR_BASE unset, using "{ZEPHYR_BASE}"')
```
Comment (gmarull) [inline scripts/ci/check_compliance.py:33]: we should in fact automatically guess Zephyr base, script is at $ZEPHYR_BASE/ci

**Review by gmarull — DISMISSED** [review]
lgtm

File: scripts/ci/check_compliance.py (hunk 2 above)
Comment (carlescufi) [reply scripts/ci/check_compliance.py:33]: We are doing exactly that a few lines above.

File: scripts/ci/check_compliance.py (hunk 1 above)
Comment (carlescufi) [reply scripts/ci/check_compliance.py:225]: Agreed. Anyway we are using `-e` now which simplifies a lot the invocation.

File: .github/workflows/compliance.yml [hunk 3]
```diff
@@ -69,8 +69,8 @@ jobs:
         # debug
         ls -la
         git log  --pretty=oneline | head -n 10
-        ./scripts/ci/check_compliance.py --annotate -m Devicetree -m Gitlint \
-        -m Identity -m Nits -m pylint -m checkpatch -m Kconfig \
+        ./scripts/ci/check_compliance.py --annotate -m DevicetreeBindings
```
Comment (nashif) [inline .github/workflows/compliance.yml]: dont you need a \ here?

Comment (carlescufi) [reply .github/workflows/compliance.yml]: I do, thanks!

**Review by nashif — DISMISSED** [review]

File: .github/workflows/compliance.yml (hunk 3 above)
Comment (nashif) [reply .github/workflows/compliance.yml]: I think you fixed this in a later commit

Comment (carlescufi) [reply .github/workflows/compliance.yml]: Hmm wait, this was fixed after in fact.

**Review by gmarull — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**